src/main/resources/static/surveyor.js

```diff
@@ -39,10 +39,219 @@ $(document).ready(function() {
+                    }));
+                }
+                // console.log(numericalRangeQuestions)
+                break
+
+            //Multiple choice questions
+            case 'multipleChoiceQuestion':
+
+                //Declare optionsJSON
+                let optionsJSON = {}
+
+                //loop through each mcq option
+                let error = false
+                $(question).find(".MCQOption").each(function () {
+                    let option = $(this).val();
+                    console.log(option);
+
+                    //Check if all the MCQ options are filled
+                    if (!option) {
+                        alert("ERROR: One of the MCQ options is empty for question: " + questionText);
+                        error = true
+                    }
+
+                    if(option in optionsJSON) {
+                        alert("ERROR: Choices for MCQ must be unique for question " + questionText)
+                        error = true
+                    }
+
+                    //Add to data structure
+                    optionsJSON[option] = 0;
+                });
+
+                // Return the error that occured within the option
+                if (error) return error
+
+                jsonQuestions.push(JSON.stringify({
+                        "question": questionText,
+                        "options": optionsJSON,
+                }));
+                    // console.log(multipleChoiceQuestions)
+                break
+
+            default:
+                alert("ERROR: Something went wrong...")
+                return true
+        }
+
+    }
+    /**
+     * Handler for AJAX post requests for regular requests, and FETCH API requests with Promises
+     * @param url
+     * @param jsonData
+     * @param returnPromise
+     * @returns Promise if indicated in the returnPromise flag, otherwise will be void
+     */
+    async function handlePostRequest(url, jsonData, returnPromise=false) {
+        //Handle POST request with ajax
+        if(returnPromise === false) {
+            $.ajax({
+                type: "POST",
+                url: url,
+                headers: {
+                    'Accept': 'application/json',
+                    'Content-Type': 'application/json'
+                },
+                data: jsonData
+            }).then(function (data) {
+                console.log("Created post at " + url + " for id " + data.id);
+            });
+        }
+        //Handle POST request with FETCH API since it uses promises
+        else{
+            const settings = {
+                method: 'POST',
+                headers: {
+                    'Accept': 'application/json',
+                    'Content-Type': 'application/json',
+                },
+                body: jsonData
+            };
+            try {
+                const fetchResponse = await fetch(url, settings);
+                return await fetchResponse.json();
```

Comment:
why is this function only returning things if return promise ===true? Also i dont even think we need this logic here. We can just use .ajax with .done and .fail